File: leetCode/fukun/101-200/lc_104.js
function TreeNode(val) {
    this.val = val;
    this.left = this.right = null;
}

const maxDepth = function (root) {
    if (!root) return 0;
    if (!root.left && !root.right) return 1;
    return Math.max(maxDepth(root.left), maxDepth(root.right)) + 1;
};

let t = new TreeNode(1);
t.left = new TreeNode(2);
t.right = new TreeNode(2);

console.log(maxDepth(t));

File: leetCode/lwc/lc_104.java
package solution2;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * @author lwc
 * @date 2019/10/26 15:49
 */
public class U104 {
    public int maxDepth(TreeNode root) {
        if(root == null)
            return 0;
        if(root.left == null && root.right == null)
            return 1;
        else
            return Math.max(maxDepth(root.left),maxDepth(root.right))+1;
    }
}
